File: _정답코드/1주차/09_완주하지_못한_선수/solution.py
def solution(participant, completion):
    freq = {}
    
    for name in participant:
        freq[name] = freq.get(name, 0) + 1
    
    for name in completion:
        freq[name] -= 1
    
    for name, count in freq.items():
        if count > 0:
            return name
    
    return ""

File: _정답코드/1주차/09_완주하지_못한_선수/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> freq = new HashMap<>();
        
        for (int i = 0; i < participant.length; i++) {
            String name = participant[i];
            freq.put(name, freq.getOrDefault(name, 0) + 1);
        }
        
        for (int i = 0; i < completion.length; i++) {
            String name = completion[i];
            freq.put(name, freq.get(name) - 1);
        }
        
        for (Map.Entry<String, Integer> entry : freq.entrySet()) {
            if (entry.getValue() > 0) {
                return entry.getKey();
            }
        }
        
        return "";
    }
}

